pkg/s2i/img/img.go

```diff
@@ -0,0 +1,74 @@
+package img
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/caicloud/ciao/pkg/types"
+)
+
+const (
+	prefix     = "kubeflow-kernel-code."
+	codeFile   = "code.py"
+	dockerFile = "Dockerfile"
+	imageOwner = "gaocegege"
+)
+
+// Client is the type for using img.
+type Client struct {
+}
+
+// New creates a new Client.
+func New() *Client {
+	return &Client{}
+}
+
+// SourceToImage converts the code to the image.
+func (c Client) SourceToImage(code string, parameter *types.Parameter) (string, error) {
+	// This is a hack to let kubernetes do not pull from docker registry.
+	imageName := fmt.Sprintf("%s:v1", filepath.Join(imageOwner, parameter.GenerateName))
+
+	dir, err := ioutil.TempDir(os.TempDir(), prefix)
+	if err != nil {
+		return "", err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(dir, codeFile), []byte(code), 0666)
+	if err != nil {
+		return "", err
+	}
+
+	if err = c.writeDockerfile(dir, parameter); err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("img", "build", "-t", imageName, dir)
```

Comment:
here we need to pre-install `img`?  what if notebook itself is running in a pod?

Reply:
Then img should be added into the image, with runc and some other deps.